unit-tests/src/test/java/com/novoda/merlin/MerlinsBeardShould.java

```diff
@@ -53,4 +54,27 @@ public void returnTrueForIsConnectedWhenNetworkConnectionIsAvailable() {
         assertThat(merlinsBeard.isConnected()).isTrue();
     }
 
-}
\ No newline at end of file
+    @Test
+    public void returnTrueForIsConnectedToWifiWhenNetworkConnectedToWifiIsConnected() {
+        when(mockConnectivityManager.getNetworkInfo(WIFI.getValue())).thenReturn(mockNetworkInfo);
+        when(mockNetworkInfo.isConnected()).thenReturn(true);
+
+        assertThat(merlinsBeard.isConnectedTo(WIFI)).isTrue();
+    }
+
+    @Test
+    public void returnFalseForIsConnectedToWifiWhenNetworkConnectedToWifiIsNotConnected() {
```

Comment:
The test does not know...

It is not that important and merged anyway.
Am 29.06.2015 15:36 schrieb "Ferran Garriga" <notifications@github.com>:

> In unit-tests/src/test/java/com/novoda/merlin/MerlinsBeardShould.java
> <https://github.com/novoda/merlin/pull/58#discussion_r33463740>:
>
> > \ No newline at end of file
> > +    @Test
> > +    public void returnTrueForIsConnectedToWifiWhenNetworkConnectedToWifiIsConnected() {
> > +        when(mockConnectivityManager.getNetworkInfo(WIFI.getValue())).thenReturn(mockNetworkInfo);
> > +        when(mockNetworkInfo.isConnected()).thenReturn(true);
> > +
> > +        assertThat(merlinsBeard.isConnectedTo(WIFI)).isTrue();
> > +    }
> > +
> > +    @Test
> > +    public void returnFalseForIsConnectedToWifiWhenNetworkConnectedToWifiIsNotConnected() {
>
> Correct, but getActiveNetworkInfo() is not invoked in the code when a
> wifi is being checked.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/novoda/merlin/pull/58/files#r33463740>.
>